mbs_results/outputs/selective_editing.py

```diff
@@ -92,31 +91,108 @@ def create_standardising_factor(
-        * previous_df[o_weight]
-        * previous_df[g_weight]
+    current_df["unit_standardising_factor"] = (
+        current_df[predicted_value]
+        * current_df[a_weight]
+        * current_df[o_weight]
+        * current_df[g_weight]
     )
 
-    previous_df["standardising_factor"] = previous_df.groupby([domain, question_no])[
+    current_df["standardising_factor"] = current_df.groupby([domain, question_no])[
         "unit_standardising_factor"
     ].transform("sum")
 
-    output_df = previous_df[
+    output_df = current_df[
         [
             period,
             reference,
             question_no,
             "standardising_factor",
             predicted_value,
             imputation_marker,
-            auxiliary_value,
         ]
     ]
 
     return output_df.reset_index(drop=True)
+
+
+def calculate_auxiliary_value(
+    dataframe: pd.DataFrame,
+    reference: str,
+    period: str,
+    question_no: str,
+    frozen_turnover: str,
+    construction_link: str,
+    imputation_class: str,
+    period_selected: int,
+) -> pd.DataFrame:
+    """
+    Returning auxiliary values for questions 40 and 49.
+    Auxiliary is frozen turnover for Q40 and
+    construction link(t-1) * frozen turnover for Q49.
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        Reference dataframe
+    reference : str
+        name of column in dataframe containing reference variable
+    period : str
+        name of column in dataframe containing period variable
+    question_no : str
+        name of column in dataframe containing question code variable
+    frozen_turnover : str
+        name of column in dataframe containing auxiliary value variable
+    construction_link : str
+        name of column in dataframe containing construction link variable
+    imputation_class : str
+        name of column in dataframe containing imputation class, where
+        there is one contruction link per imputation_class and period
+    calc_period : int
+        period to take the weights for estimation of standardising factor in
+        the format yyyymm
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with calculated auxiliary_value and reference, period
+        and question_no in order to merge back onto selective editing
+        questions output
+
+    """
+    period_selected = pd.to_datetime(period_selected, format="%Y%m")
+    dataframe[period] = convert_column_to_datetime(dataframe[period])
+    current_df = dataframe[(dataframe[period] == period_selected)]
+
+    q40 = current_df[current_df[question_no] == 40]
+    q49 = current_df[current_df[question_no] == 49]
+
+    q40["auxiliary_value"] = q40[frozen_turnover]
+
+    previous_period = period_selected - pd.DateOffset(months=1)
+    prev_const_link = (
+        dataframe[
+            (dataframe[period] == previous_period) & (dataframe[question_no] == 49)
+        ][[imputation_class, construction_link, question_no]]
+        .drop_duplicates()
```

Comment:
maybe we should look for duplicates in imputation_class and question_no as a quick check?